MetaMask/metamask-extension
# Replace ActionableMessage components with BannerAlerts in SIWE Sign-in with Ethereum page

## Explanation

Fixes #18206

- .mm-icon Icon component uses a <span> element which was not supported by ActionableMessage
- Replace SIWE ActionableMessage usage with BannerAlert
  - text in alerts are slightly larger due to the change
- Add `@deprecate` jsdoc comment to ActionableMessage 

## Screenshots/Screencaps



### Before (v10.26.1)
(actually, the icon locations were temp fixed in this hotfix #18200 just before this PR.)
<img width="373" alt="Screen Shot 2023-03-16 at 4 49 31 PM" src="https://user-images.githubusercontent.com/20778143/225777090-ad46bb50-a07b-4553-bf51-a52dd33d5f29.png">
<img width="446" alt="Screen Shot 2023-03-17 at 11 07 27 AM" src="https://user-images.githubusercontent.com/20778143/225986296-31daf9e2-50dd-468f-9daf-64381e63024c.png">


### After
<img width="381" alt="Screen Shot 2023-03-17 at 10 31 51 AM" src="https://user-images.githubusercontent.com/20778143/225977356-5577e289-c03f-404e-b337-04aeaebbfffe.png">
<img width="376" alt="Screen Shot 2023-03-17 at 11 16 44 AM" src="https://user-images.githubusercontent.com/20778143/225986250-1cff6278-ec6a-4431-a667-a93799326f21.png">



## Manual Testing Steps

Login to the extension
Navigate to the test dapp
Click Sign In With Ethereum (Bad Account)
Scroll to the bottom and observe the warning
Check other ActionableMessage components



## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
